pkg/extensions/sync/sync.go

```diff
@@ -377,6 +382,8 @@ func syncRegistry(regCfg RegistryConfig, storeController storage.StoreController
 			return err
 		}
 
+		defer os.RemoveAll(path.Join(imageStore.RootDir(), imageName, SyncBlobUploadDir, uuid))
```

Comment:
Any particular reason we are suing a new var on line #372 (why not assigning directly on line #371 the upstreamRef if we are not using ref at all inside the for block)

Reply:
Yes, because I'm using the variable in function literal which I'm passing to retry.RetryIfNecessary(). You can't use range scoped vars in function literals. 
`
func() error {
			_, err = copy.Image(context.Background(), policyCtx, localRef, upstreamRef, &options)
			return err
		}
`